fix(content-explorer): Fix color contrast issues

tjuanitas opened the pull request:
The following changes have been consulted with design:

1. Increase icon color of Grid/List View button (`ViewModeChangeButton`). This aligns with color used in main web app.

**Before**
<img width="145" alt="Screen Shot 2023-01-25 at 10 03 45 AM" src="https://user-images.githubusercontent.com/7311041/214645862-6a7b00a6-9caa-4041-b62f-5923dec44809.png">
**After**
<img width="150" alt="Screen Shot 2023-01-25 at 10 04 40 AM" src="https://user-images.githubusercontent.com/7311041/214646008-a3acf30d-fb7e-49e0-8bc7-a7388d500587.png">

~~2. Improve contrast on "More Options" button by adding white background to button and darker icon color. This is a similar pattern from Admin Console~~

~~**Before**
<img width="224" alt="Screen Shot 2023-01-25 at 10 06 24 AM" src="https://user-images.githubusercontent.com/7311041/214646792-61fc6ec3-98fb-4264-9b71-6e88b573f049.png">
**After**
<img width="236" alt="Screen Shot 2023-01-25 at 10 06 43 AM" src="https://user-images.githubusercontent.com/7311041/214646742-3c2313e4-a2dd-4866-9b12-8fcfa3492109.png">~~

3. Change color of header columns and add focus outline. Closer aligns to main web app

**Before**
<img width="244" alt="Screen Shot 2023-01-25 at 10 09 30 AM" src="https://user-images.githubusercontent.com/7311041/214647098-c1e05535-4c33-408e-bc83-076aee50ee21.png">
**After**
<img width="244" alt="Screen Shot 2023-01-25 at 10 09 45 AM" src="https://user-images.githubusercontent.com/7311041/214647141-1ecbb870-e5c7-49cb-b09b-f84ed6c865ef.png">

---

Edit:

- Updated "Sort" and "Add" icon colors to match "Grid/List" icons

**Before**
<img width="838" alt="Screen Shot 2023-01-26 at 7 58 32 PM" src="https://user-images.githubusercontent.com/7311041/215006807-c9e069bf-7c5f-4497-9da3-0a53f91bd85c.png">
**After**
<img width="892" alt="Screen Shot 2023-01-26 at 7 56 41 PM" src="https://user-images.githubusercontent.com/7311041/215006810-ce7d3c38-63d5-4f99-948e-e4d9dfdf05e3.png">

- Updated border and icon color for "More Options" button states

**Hover State - Before**
<img width="676" alt="Screen Shot 2023-01-26 at 8 03 32 PM" src="https://user-images.githubusercontent.com/7311041/215007240-9cb5a9ec-f5b7-448f-82ef-1fdab788e2c7.png">

**Hover State - After**
<img width="650" alt="Screen Shot 2023-01-26 at 8 04 27 PM" src="https://user-images.githubusercontent.com/7311041/215007279-a0092089-eddb-45e9-8893-4b56cf48fe31.png">

**Focus State - Before**
<img width="650" alt="Screen Shot 2023-01-26 at 8 03 47 PM" src="https://user-images.githubusercontent.com/7311041/215007300-9679ad80-20f7-4fb8-95ee-e0903b91d3cb.png">

**Focus State - After**
<img width="696" alt="Screen Shot 2023-01-26 at 8 04 35 PM" src="https://user-images.githubusercontent.com/7311041/215007314-d1c2cedc-1d42-4d99-80af-8b73750cf99c.png">

jstoffan reviewed (commented):
@tjuanitas, do the border colors need to be updated, as well? They seem to be the same color as the icons in the current design.

tjuanitas commented:
@jstoffan I'll double check all the colors with my designer. The `ViewModeChangeButton` border seems okay to me because the web app also uses the default border-color for icon buttons but the "More Options" button seems off.

tjuanitas commented on src/elements/content-explorer/ItemList.scss:
```diff
@@ -56,6 +57,10 @@
         }
 
+        .bce-btn-more-options {
+            border-color: $grey-blue;
```
using a deprecated color to match current item row border-color

tjuanitas commented on src/elements/common/sub-header/AddButton.js:
```diff
@@ -10,12 +10,13 @@ import Button from '../../../components/button';
     <Tooltip text={<FormattedMessage {...messages.add} />}>
         <Button className="be-btn-add" aria-label={messages.add.defaultMessage} type="button" {...props}>
-            <IconAddThin />
+            <IconAddThin color={bdlGray62} />
```
legacy icon colors are modified through props

tjuanitas commented:
@jstoffan We ended up making a couple more changes to align the color of the icons on the sub-header component of Explorer and Picker, and also updated the border color of the "More Options" button. Added more screenshots to the description.
